proposal.rst

```diff
@@ -197,45 +233,88 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+
+I am still trying to see if we can utilize the N-gram algorithm to help on this task.I made a google `doc <https://docs.google.com/document/d/1yXdnAhtNKWcODLLBy2hB2pluQ2nr3EfZJP2_F1XIpDs/edit
+/>`_ that I keep updated with results I reached so far.
+
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+There's approach addressed in a `patch <https://oligarchy.co.uk/xapian/patches/docid-ranges-in-matcher.patch/>`_ by Olly,
+that make use of the fact that we have the ids for the first and last documents now.
+Example: for the "MultiAndPostList", we would mutiply the probability of each term independantly, and we would do that by
+divide the term frequnecy(from term_freq_est() function) by the database size. But now we can get the range from the first docid to the last docid
+(last - first +1), and we use that range instead of the whole database size.
```

Comment:
So AND and OR are mostly done already, but what does the calculation look like for some of the other operators?  For example `AND_NOT`?